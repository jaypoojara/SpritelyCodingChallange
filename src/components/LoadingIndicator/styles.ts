import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

const styles = StyleSheet.create({
  loadingView: {
    height: 45,
    width: 45,
    borderRadius: 45,
    backgroundColor: colors.white,
    justifyContent: 'center',
    alignItems: 'center',
    elevation: 5,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0.3,
    shadowRadius: 10,
  },
});

export default styles;
