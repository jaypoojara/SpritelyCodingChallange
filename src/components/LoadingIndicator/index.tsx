import { View, ActivityIndicator } from 'react-native';
import React from 'react';
import colors from '../../utils/colors';
import styles from './styles';

export default function LoadingIndicator() {
    return (
        <View testID="activity-loader-id" style={styles.loadingView}>
            <ActivityIndicator
                color={colors.primary}
            />
        </View>
    );
}