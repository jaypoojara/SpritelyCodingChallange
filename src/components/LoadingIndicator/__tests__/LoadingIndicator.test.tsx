import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LoadingIndicator from '..';

test('renders correctly', () => {
  const element = renderer.create(<LoadingIndicator />).toJSON();
  expect(element).toMatchSnapshot();
});