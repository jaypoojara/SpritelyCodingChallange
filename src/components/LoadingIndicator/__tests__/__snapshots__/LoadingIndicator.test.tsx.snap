// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 45,
      "elevation": 5,
      "height": 45,
      "justifyContent": "center",
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 10,
      "width": 45,
    }
  }
  testID="activity-loader-id"
>
  <ActivityIndicator
    color="#003A67"
  />
</View>
`;
